src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -5339,6 +5339,66 @@ void MacroAssembler::safepoint_isb() {
 #endif
 }
 
+Assembler::SIMD_Arrangement elemBytes_to_Arrangement(int esize, bool isQ) {
+  switch(esize) {
+    case 1:
+      return isQ ? Assembler::T16B : Assembler::T8B;
+    case 2:
+      return isQ ? Assembler::T8H : Assembler::T4H;
+    case 4:
+      return isQ ? Assembler::T4S : Assembler::T2S;
+    case 8:
+      return isQ ? Assembler::T2D : Assembler::T1D;
+    default:
+      assert(false, "unsupported");
+      ShouldNotReachHere();
+  }
+  return Assembler::INVALID_ARRANGEMENT;
+}
+
+void MacroAssembler::neon_compare(FloatRegister dst, BasicType bt, FloatRegister src1,
+                                  FloatRegister src2, int cond, bool isQ) {
+  SIMD_Arrangement size = elemBytes_to_Arrangement(type2aelembytes(bt), isQ);
+  if (bt == T_FLOAT || bt == T_DOUBLE) {
+    switch (cond) {
+      case BoolTest::eq: fcmeq(dst, size, src1, src2); break;
+      case BoolTest::ne: {
+        fcmeq(dst, size, src1, src2);
+        notr(dst, T16B, dst);
+        break;
+      }
+      case BoolTest::ge: fcmge(dst, size, src1, src2); break;
+      case BoolTest::gt: fcmgt(dst, size, src1, src2); break;
+      case BoolTest::le: fcmge(dst, size, src2, src1); break;
+      case BoolTest::lt: fcmgt(dst, size, src2, src1); break;
```

Comment:
Are you sure about these two lines?

Reply:
Sorry, I see what's happening: the operands are reversed. This is tricky code.

Reply:
As A64-SIMD doesn't have `fcmlt(register)` and `fcmle(register)` instructions, I think it's okay to use `fcmgt` and `fcmge` with reversing the argument. But I can't locate the initial patch adding them.

Reply:
Yes, I see now.